SteamKit2/SteamKit2/Steam/Discovery/SmartCMServerList.cs

```diff
@@ -300,14 +256,30 @@ public IPEndPoint GetNextServerCandidate()
         }
 
         /// <summary>
+        /// Get the next server in the list.
+        /// </summary>
+        /// <returns>An <see cref="System.Net.IPEndPoint"/>, or null if the list is empty.</returns>
+        public Task<IPEndPoint> GetNextServerCandidateAsync()
+        {
+            StartFetchingServers();
+            return listTask.ContinueWith( t =>
+            {
+                return GetNextServerCandidate();
```

Comment:
This will end up calling `StartFetchingServers` twice, and locking unneccesarily. Seems wasteful.